portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,25 +32,33 @@
 /** Servlet that handles user comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> comments = new ArrayList<>();
   private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final String dataKind = "Comment";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> comments = new ArrayList<>();
+    Query query = new Query(dataKind).addSort("timestamp", SortDirection.ASCENDING);;
+    PreparedQuery results = datastore.prepare(query);
+    
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
+      comments.add(getCommentStatement(name, comment, timestamp));
```

Comment:
you can create a function like buildComment(Entity);

Reply:
Done.